module3/lessons/intro_to_oauth.md

```diff
@@ -5,12 +5,9 @@ length: 180
 tags: rails, security, authentication, OAuth
 ---
 
-This lesson plan was last tested to work with Rails 5.2.6 and Ruby 2.5.3
-
-
 ## Historical Context for this lesson plan
 
-We used to teach this class using a gem since that’s how most of us do it on the job. We saw a big difference in understanding for students who were able to hand roll the handshake and those that used a gem. The other interesting thing we uncovered was many students struggled troubleshooting the gem if they didn’t understand the process. This lesson is built showing how to hand roll the OAuth process. The expectation students should try to use a gem on their projects. The big reason for using a gem is it’s much easier to test and most struggle with testing external APIs.
+We have taught this class many different ways -- hand rolling the handshake and using a gem. While hand rolling the handshake provides the most in-depth learning for the OAuth flow, it is a level of detail that is often not as applicable on the job. However, it's important to understand the OAuth handshake at a high level in order to better troublshoot the gem. This lesson includes a section showing how to hand roll the OAuth process if students are interested. However, performing the handshake manually is not a learning goal for this lesson. The expectation is for students to use a gem on their projects. The big reason for using a gem is it’s much easier to test and most struggle with testing external APIs.
```

Comment:
Hand rolling an OAuth handshake seems like a good thing for students to know, but I agree that the lesson time is likely better spent introducing the basics of what they'll need for Consultancy (omniauth). 
Maybe in the future we could experiment with assigning homework/class prep that explains the more involved OAuth approach. I wonder if that might generate conversation/questions and be valuable context for what Omniauth provides, without making a hand rolled implementation too much of a focus during the lesson.

Reply:
Thanks, Jamison! I totally agree. I'm thinking we keep the lesson plan section that goes through the hand rolling process but I'm hoping that adding this blurb at the top reduces any confusion that's come up in the past with students reading it and then questioning why we're not hand rolling in class. 

I love the idea of doing prework. I didn't do it this inning because it seemed tight with inception deliverables due and the students' job shadow. But maybe next time we can wiggle the calendar to make that work. Thanks for the idea!